src/main/scala/io/circe/yaml/snake/CirceJsonRepresenter.scala

```diff
@@ -0,0 +1,107 @@
+package io.circe.yaml.snake
+
+import io.circe.{Json, JsonDouble, JsonLong}
+import io.circe.Json._
+import io.circe.yaml.snake.CirceJsonRepresenter.RepresentCirceJson
+import org.yaml.snakeyaml.nodes.{Node, Tag}
+import org.yaml.snakeyaml.representer.{Represent, Representer}
+
+import scala.collection.JavaConverters._
+
+/**
+  * Represents [[Json]] as Yaml [[Node]]s.
+  *
+  * @note This extends [[Representer]] because [[org.yaml.snakeyaml.Yaml]] does not accept the
+  *       [[org.yaml.snakeyaml.representer.BaseRepresenter]] class. It only accepts this specific class.
+  */
+class CirceJsonRepresenter extends Representer {
+
+  // Add the only necessary representer to the map and remove all others
+  this.representers.clear()
+  // Need to represent field names as strings
+  this.representers.put(classOf[String], RepresentFieldName)
+  this.representers.put(null, representUnexpected)
+  this.multiRepresenters.clear()
+  this.multiRepresenters.put(classOf[Json], representJson)
+  this.nullRepresenter = representJson
+
+  def representJson: RepresentCirceJson = StandardRepresentCirceJson
+  def representUnexpected: RepresentUnexpected = RepresentUnexpected
+
+  object RepresentFieldName extends RepresentFieldName
+  class RepresentFieldName extends RepresentString
+
+  object RepresentUnexpected extends RepresentUnexpected
+  class RepresentUnexpected extends Represent {
+
+    def displayData(data: Any): String = {
+      val dataAsString = data.toString
+      if (dataAsString.length <= 500) dataAsString
+      else dataAsString.substring(0, 499) + "…"
+    }
+
+    override def representData(data: Any): Node = {
+      val className = data.getClass.getName
+      throw new IllegalArgumentException(
+        s"Cannot represent an instance of '$className' with value '${displayData(data)}'. " +
+          s"Please convert this to ${classOf[Json].getName} first using io.circe.Decoder[$className].decode"
+      )
+    }
+  }
+
+  object StandardRepresentCirceJson extends StandardRepresentCirceJson
+  class StandardRepresentCirceJson extends RepresentCirceJson {
+
+    override def represent(json: Json): Node = json match {
```

Comment:
Yea, that's true. I was using `.fold` at first, but I needed access to `JsonDouble` to handle the infinity case and `.toBiggerDecimal` to handle determining when a big number is a floating point or integer, and these are both `private[circe]` as well.

@travisbrown what would you recommend as the way to detect between these 3 cases:
- finite decimal
- finite integral
- positive infinity
- negative infinity
- NaN

The parser and the tagging of node types in Yaml requires that we appropriately label a node with the correct type. So `1.0` should be a `!!float` and not an `!!int` and `.inf` should be `Double.PositiveInfinity`, etc.